spec/models/prison_spec.rb

```diff
@@ -1,6 +1,21 @@
 require 'rails_helper'
 
 RSpec.describe Prison, type: :model do
+  describe '#active' do
+    before do
+      create(:allocation, prison: p1.code)
+      create(:allocation, prison: p1.code)
+      create(:allocation, prison: p2.code)
+    end
+
+    let(:p1) { build(:prison) }
+    let(:p2) { build(:prison) }
+
+    it 'only lists prisons with allocations' do
+      expect(described_class.active.map(&:code)).to match_array [p1, p2].map(&:code)
+    end
+  end
+
```

Comment:
Can I check, Is the definition of 'active' purely that a prison has offenders in it who have been allocated a pom or com? Where a prison has no allocations taking place, this prison is see as not active and therefore doesn't want to receive an automatic email? Is that right?

Reply:
It might be nice to have a third prison here (`p3`?) created but not allocated to, to show that it doesn't come back in the active list.

Something like:

```
let!(:p3) { build(:prison) }
```

and then

```
expect(described_class.active.map(&:code)).not_to include p3.code
```

Reply:
@ollietreend Prisons already exist. building them doesn't do anything other than make them accessible. So the test already shows that only the ones with allocations are retrieved.